### PR TITLE
Convert SSL Observatory to use getBrowserForChannel.

### LINKED ISSUE #786: Firefox 37: TypeError: this.HTTPSEverywhere.getWindowForChannel is not a function ssl-observatory.js:819:24
With the current https-everywhere-5.0development.1.xpi on Firefox Nightly 37, I see the following error logged (even if e10s is disabled):

`TypeError: this.HTTPSEverywhere.getWindowForChannel is not a function ssl-observatory.js:819:24`

### PR DESCRIPTION
**Pull request by jsha**
Fixes #786

### CODE REVIEW

File: src/components/ssl-observatory.js [hunk 1]
```diff
@@ -611,7 +611,11 @@ SSLObservatory.prototype = {
     var HTTPSEverywhere = CC["@eff.org/https-everywhere;1"]
                             .getService(Components.interfaces.nsISupports)
                             .wrappedJSObject;
-    var win = channel ? HTTPSEverywhere.getWindowForChannel(channel) : null;
+    var win = null
+    if (channel) {
+      var browser = this.HTTPSEverywhere.getBrowserForChannel(channel);
+      var win = browser.contentWindow;
```
Comment (emanueLczirai) [inline src/components/ssl-observatory.js]: This line gives `browser is null` ssl-observatory.js:624
Firefox 34.0.5 linux, https-everywhere latest master (currently this commit is latest: d278a6dff48e75fe6efc12df724f7a318ebbf3ac )
Seems to happen on browser startup(and restores tabs). I don't know what to make of it.

Comment (jsha) [inline src/components/ssl-observatory.js]: I fixed the by adding a check for browser being defined.

**Comment by jsha** [comment]
ping @cooperq for review.

**Comment by emanueLczirai** [comment]
tl;dr: works ok for me.

----
Seems to be working alright for me. I get no warnings(aside from the unrelated deprecated nsIJSON.encode) or errors. Thanks.


As an aside, I want to mention that when `observatory.eff.org` is blocked (in my case, in /etc/hosts made to point to 127.0.0.3 where there is nothing listening on any port there; for tests) then `req.status` is `0` , req.responseText is empty and thus this shows on firefox console: `SSL Observatory WARN: Cert submission failure 0:`.

(note to self) This helped me increase Firefox's WebConsole log level:
https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/console#Setting_the_Logging_Level
Basically, about:config New->String `extensions.sdk.console.logLevel` set value to `all`.
However, for https-everywhere logs to show up I had to change `extensions.https_everywhere.LogLevel` from `5` to `1` to see them all (and the above WebConsole log level changes are not required)

File: src/components/ssl-observatory.js [hunk 2]
```diff
@@ -618,7 +618,13 @@ SSLObservatory.prototype = {
     var HTTPSEverywhere = CC["@eff.org/https-everywhere;1"]
                             .getService(Components.interfaces.nsISupports)
                             .wrappedJSObject;
-    var win = channel ? HTTPSEverywhere.getWindowForChannel(channel) : null;
+    var win = null
```
Comment (cooperq) [inline src/components/ssl-observatory.js]: Nit: Needs a semicolon

**Comment by cooperq** [comment]
Has this been tested on firefox mobile yet?

**Comment by jsha** [comment]
Not yet, will do. Thanks for the reminder!

**Comment by Yoyo117** [comment]
Will this get merged for the next dev-version? This is really spamming my Browser-Console....

**Comment by jsha** [comment]
Yep!

**Comment by jsha** [comment]
Ok, tested on Firefox mobile and works well.
